### PR TITLE
Docs ### Interacting with contracts deployed on Mainnet

### PR DESCRIPTION
**Pull request by dabuchera**
Hi,

Was refining a bit the docs about the interacting with contract deployed on Mainnet. 
Mainly elaborating with a short example but also changing the syntax of adding requirements. 

Best,

### CODE REVIEW

**Review by obycode — CHANGES_REQUESTED** [review]
Thanks @dabuchera for this PR! I've added a few comments and suggestions.

File: README.md [hunk 1]
```diff
@@ -451,7 +450,115 @@ From there, clarinet will be able to resolve the `contract-call?` statements inv
...
+
+```clarity
+(impl-trait 'SP2PABAF9FTAJYNFZH93XENAJ8FVY99RRM50D2JG9.nft-trait.nft-trait)
+```
+
+Similar to the deployment on Mainnet, the requirements should be listed in the manifest `Clarinet.toml`
+
+Dependencies from **external** contracts should be set in `[[project.requirements]]`
+
+Dependencies from **internal** contracts should be set in `depends_on`
```
[the remark this answers is not in the document]
Comment (dabuchera) [reply README.md:479]: I once speculated that, but was not sure. I will document that as deprecated but often seen in tutorials and documentation.

File: README.md [hunk 2]
```diff
@@ -451,7 +450,115 @@ From there, clarinet will be able to resolve the `contract-call?` statements inv
...
+)
+```
+
+### External Deployer 
+
+```clarity
+(impl-trait 'SP2PABAF9FTAJYNFZH93XENAJ8FVY99RRM50D2JG9.nft-trait.nft-trait)
+```
+
+Similar to the deployment on Mainnet, the requirements should be listed in the manifest `Clarinet.toml`
```
[the remark this answers is not in the document]
Comment (dabuchera) [reply README.md:475]: Agree

File: README.md [hunk 3]
```diff
@@ -451,7 +450,115 @@ From there, clarinet will be able to resolve the `contract-call?` statements inv
...
+(define-read-only (get-token-uri (token-id uint))
+  (if (< token-id u5001)
+    (ok (some (concat (concat (var-get ipfs-root) (unwrap-panic (contract-call? .conversion lookup token-id))) ".json")))
+    (ok (some (concat (concat (var-get ipfs-root) (unwrap-panic (contract-call? .conversion-v2 lookup (- token-id u5001)))) ".json")))
+    )
+)
+```
+
+### External Deployer 
+
```
[the remark this answers is not in the document]
Comment (dabuchera) [reply README.md:470]: Agree

File: README.md [hunk 4]
```diff
@@ -451,7 +450,115 @@ From there, clarinet will be able to resolve the `contract-call?` statements inv
...
 When deploying your protocol to Devnet / Testnet, for the contracts involving requirements, the setting `remap_requirements` in your deployment plans must be set.
 
-Before Devnet / Testnet deployments, your contracts will be automatically remapped on the fly to point to the duplicated requirements deployed by an account that you control.
+As an step-by-step example we use here the following contract **bitcoin-whales** you can find [here](https://explorer.stacks.co/txid/SP2KAF9RF86PVX3NEE27DFV1CQX0T4WGR41X3S45C.bitcoin-whales?chain=mainnet)
+
+If you examine this, you will see that we have 3 different dependencies within the contract. Two from the **internal**
+ deployer, one reffering to an contract deployed by an **external** deployer
+
+### Same Deployer 
+
```
[the remark this answers is not in the document]
Comment (dabuchera) [reply README.md:459]: Agree

File: README.md [hunk 5]
```diff
@@ -451,7 +450,117 @@ From there, clarinet will be able to resolve the `contract-call?` statements inv
...
+contract_id = "SP2PABAF9FTAJYNFZH93XENAJ8FVY99RRM50D2JG9.nft-trait"
+
+boot_contracts = ["pox", "costs-v2", "bns"]
+
+[project.cache_location]
+path = ".requirements"
+
+[contracts.bitcoin-whales]
+path = "contracts/bitcoin-whales.clar"
+#depends_om = ["conversion","conversion-v2"] #optional deprecated
```
Comment (obycode) [inline README.md:497]:
```suggestion
# depends_on = ["conversion","conversion-v2"] # no longer needed, ignored if provided
```

**Review by obycode — CHANGES_REQUESTED** [review]
Thanks for the updates! Just one minor change requested.

**Review by obycode — APPROVED** [review]
Thanks!

**Pull request merged**